src/goto-instrument/contracts/assigns.cpp

```diff
@@ -112,14 +113,32 @@ void assigns_clauset::add_target(const exprt &target_expr)
 
 void assigns_clauset::remove_target(const exprt &target_expr)
 {
-  targets.erase(targett(*this, targett::normalize(target_expr)));
+  write_set.erase(targett(*this, targett::normalize(target_expr)));
```

Comment:
Same comment -- constructor calls `normalize` already. This should remain unchanged.

Reply:
Correct, remove it.